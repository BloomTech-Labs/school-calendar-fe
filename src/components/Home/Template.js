import React, {setState} from 'react';

//opens overlay for user to pick dates in calendar.
//user clicks on dates, which get added to list
//when user clicks 'Apply', create event on each day in list
const applyTemplate = () => {
  console.log('select days on calendar you wish to add this event to');
};

const Template = ({starttime, endtime, summary, description}) => {
  
  return (
    <div>
      {starttime}
      {endtime}
      {summary}
      {description}
      <button onClick={() => applyTemplate()}> + </button>
    </div>
  );
};
export default Template;
