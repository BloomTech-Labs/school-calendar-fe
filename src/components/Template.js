import React, {useState, useReducer} from 'react';
import {
  Flex,
  ButtonGroup,
  Button,
  Heading,
  IconButton
} from '@chakra-ui/core';

//TODO LINK UP THE DAYS.JS TRIGGER TO MATCH THE TEMPLATE TOGGLE FROM A STATE WIDE SELECTION.


const Template = ({
  id,
  starttime,
  endtime,
  summary,
  description,
  selected,
  setSelected,
  templateFormOpen,
  setTemplateFormOpen,
  applyTemplate,
  handleDelete
}) => {



  const openTemplate = () => {
    setTemplateFormOpen(!templateFormOpen);
    setToggledTemplate(!toggledTemplate);
    setSelected([]);
  };

  const [toggledTemplate, setToggledTemplate] = useState(false)
  
  return (
    <Flex direction="column" align="center" justify="center" my={2}>
      <Heading fontSize="sm" fontWeight="normal">
        {summary}
      </Heading>
      <Heading fontSize="sm" fontWeight="normal">
        {starttime}-{endtime}
      </Heading>
      <Flex>
        <ButtonGroup spacing={4}>
          <Button size="sm" variantColor="blue" onClick={() => openTemplate()}>
            Choose Dates
          </Button>
          <IconButton
            variantColor="red"
            aria-label="Delete"
            size="sm"
            icon="close"
            onClick={() => handleDelete(id)}
          />
        </ButtonGroup>
      </Flex>

      {toggledTemplate && (
        <button
          onClick={() =>
            applyTemplate(summary, description, starttime, endtime, selected)
          }
        >
          Apply Template
        </button>
      )}
    </Flex>
  );
};

export default Template;
