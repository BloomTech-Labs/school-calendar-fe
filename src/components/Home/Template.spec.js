import React from 'react';
import { shallow } from 'enzyme';
import Template from './Template';


describe('Template', () => {
  it('should render correctly', () => {
    const component = shallow(<Template />);
  
    expect(component).toMatchSnapshot();
  });

  
});