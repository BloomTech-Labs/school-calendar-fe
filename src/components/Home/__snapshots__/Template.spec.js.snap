// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template should render correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
     + 
  </button>
</div>
`;
